eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py

```diff
@@ -170,120 +203,145 @@ def query_mongo(mongo_source, filter_criteria, metric):
-                                      f"{new_clustered_current_rs}," \
-                                      f"{new_clustered_current_rs})"
+                           f"values ({taxid}, '{scientific_name}', '{asm}', '{folder}', {release_version}, "
             insert_query = f"{insert_query} {insert_query_values}"
             logger.info(insert_query)
             execute_query(metadata_connection_handle, insert_query)
 
-        # get assemblies in from previous releases not in current release
-        assemblies_in_logs = ",".join(f"'{a}'" for a in ranges_per_assembly.keys())
-        query_missing_assemblies_stats = f"select * " \
-                                         f"from {assembly_table_name} " \
-                                         f"where release_version = {release_version}-1 " \
-                                         f"and assembly_accession not in ({assemblies_in_logs});"
-        logger.info(query_missing_assemblies_stats)
-        missing_assemblies_stats = get_all_results_for_query(metadata_connection_handle, query_missing_assemblies_stats)
-        for assembly_stats in missing_assemblies_stats:
-            taxonomy_id = assembly_stats[0]
-            scientific_name = assembly_stats[1]
-            assembly_accession = assembly_stats[2]
-            release_folder = assembly_stats[3]
-            current_rs = assembly_stats[5]
-            multi_mapped_rs = assembly_stats[6]
-            merged_rs = assembly_stats[7]
-            deprecated_rs = assembly_stats[8]
-            merged_deprecated_rs = assembly_stats[9]
+
+def fill_data_from_previous_release(metadata_connection_handle, ranges_per_assembly, release_version):
+    """Insert metrics for taxonomies and assemblies *not* processed during the current release, but present in a
+    previous release."""
+    assemblies_in_logs = {a for a in ranges_per_assembly.keys()}
+    asm_tax_in_logs = {(asm, tax) for asm in assemblies_in_logs for tax, _ in ranges_per_assembly[asm]['species_info']}
+    previous_release_stats = get_all_results_for_query(metadata_connection_handle,
+                                                       f"select * from {assembly_table_name} "
+                                                       f"where release_version = {release_version-1}")
+    for previous_assembly_stats in previous_release_stats:
+        taxonomy_id = previous_assembly_stats[0]
+        scientific_name = previous_assembly_stats[1]
+        assembly_accession = previous_assembly_stats[2]
+        release_folder = previous_assembly_stats[3]
+
+        # If the complete taxonomy/assembly pair is already present in the current release, do nothing
+        if (taxonomy_id, assembly_accession) in asm_tax_in_logs:
+            continue
+
+        # If the assembly is present in the current release but not with this taxonomy,
+        # copy this release's metrics for this assembly.
+        if assembly_accession in assemblies_in_logs:
```

Comment:
And here we're retrieving the updated taxonomies populated in `fill_data_for_current_release` and using them as template.